src/gt4py/next/iterator/transforms/fuse_as_fieldop.py

```diff
@@ -40,10 +41,8 @@ def _merge_arguments(
 ) -> dict[str, itir.Expr]:
     new_args = {**args1}
```

Comment:
Probably unrelated to this PR, but why are they called `args1` and `arg2`? Shouldn't the second argument be called `args2`?